File: math/number-theory/is_even.ts
/**
 * Copyright https://github.com/microwind
 * Check if number is even/odd
 */

function isEven(n: number): boolean {
    return n % 2 === 0;
}

function main(): void {
    const testNumbers = [2, 3, 4, 5, 10, 15];
    for (const n of testNumbers) {
        console.log(`${n} is ${isEven(n) ? 'even' : 'odd'}`);
    }
}

main();
export { isEven };

File: math/number-theory/is_even.rs
// Copyright https://github.com/microwind
// Check if number is even/odd

fn is_even(n: i32) -> bool {
    n % 2 == 0
}

fn main() {
    let test_numbers = [2, 3, 4, 5, 10, 15];
    for n in test_numbers {
        println!("{} is {}", n, if is_even(n) { "even" } else { "odd" });
    }
}
